runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java

```diff
@@ -167,6 +182,50 @@ public void onChatMessage(ChatMessage chatMessage)
 		}
 	}
 
+	@Subscribe
+	public void onMenuOpened(MenuOpened event)
+	{
+		if (event.getMenuEntries().length < 2 || !config.copyToClipboard())
+		{
+			return;
+		}
+
+		// Use second entry as first one can be walk here with transparent chatbox
+		final MenuEntry entry = event.getMenuEntries()[event.getMenuEntries().length - 2];
+
+		final int groupId = TO_GROUP(entry.getParam1());
+		final int childId = TO_CHILD(entry.getParam1());
+
+		if (groupId != WidgetInfo.CHATBOX.getGroupId())
+		{
+			return;
+		}
+
+		final Widget widget = client.getWidget(groupId, childId);
+		final Widget parent = widget.getParent();
+		final Widget[] dynamicChildren = parent.getDynamicChildren();
+
+		// Get child id of first chat message static child so we can substract this offset to link to dynamic child
+		// later
+		final int first = TO_CHILD(parent.getStaticChildren()[0].getId());
+
+		// Convert current message static widget id to dynamic widget id of message node with message contents
+		final int dynamicChildId = (childId - first) * 2 + 1;
```

Comment:
also an explanation of what the widget tree looks like for this to work would be nice